homework9/src/pages/Courses/Courses.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useCallback, useEffect, useState } from "react";
+import { connect } from "react-redux";
+import { createStructuredSelector } from "reselect";
+import BounceLoader from "react-spinners/BounceLoader";
+import getSymbolFromCurrency from "currency-symbol-map";
+
+import { getCoursesAction } from "../../resources/course/course.actions";
+import { selectCourses } from "../../resources/course/course.selectors";
+
+import Button from "../../components/Button";
+
+import "./courses.styles.css";
+
+const Courses = ({ onGetCoursesAction, courses }) => {
+  const [loading, setLoading] = useState(true);
+  const [errorMsg, setErrorMsg] = useState(null);
+
+  const fetchCourses = useCallback(async () => {
+    setErrorMsg(null);
+    setLoading(true);
+    const res = await onGetCoursesAction();
+    if (res.type === "success") {
+      setErrorMsg(null);
+      setLoading(false);
+    } else {
+      setErrorMsg("No course found");
```

Comment:
u can set up that message on redux level. Otherwise u should track it on every single component
but on redux level only one set up

Reply:
for clear message as on 23 line u can do it  in redux as well